src/devtools/client/webconsole/components/Search/useSearch.ts

```diff
@@ -66,16 +67,25 @@ export default function useSearch<Item>(
   // this keeps the search text input updating quick and feeling responsive to user input.
   const deferredQuery = useDeferredValue(state.query);
 
-  const prevItemsRef = useRef<Item[] | null>(null);
-  const prevQueryRef = useRef<string | null>(null);
+  // TRICKY
+  // Store previous values in State; do not store them in a Ref.
+  // It's technically not safe to read or write to a Ref during render.
+  // If React bails out before committing an update, the Ref value will have already been mutated when the component later renders.
+  // That means the render-phase state update (re-running search) will be skipped.
+  const [prevValues, setPrevValues] = useState<{
+    items: Item[] | null;
+    query: string | null;
+  }>({
+    items: null,
+    query: null,
+  });
```

Comment:
Noticed an unsafe side effect (reading/writing to a ref during render) while working on the new Console's search so I fixed it here too.